filters/include/pcl/filters/impl/grid_minimum.hpp

```diff
@@ -117,21 +118,19 @@ pcl::GridMinimum<PointT>::applyFilterIndices (std::vector<int> &indices)
   // First pass: go over all points and insert them into the index_vector vector
   // with calculated idx. Points with the same idx value will contribute to the
   // same point of resulting CloudPoint
-  for (std::vector<int>::const_iterator it = indices_->begin (); it != indices_->end (); ++it)
+  for (const auto& index : (*indices_))
   {
     if (!input_->is_dense)
       // Check if the point is invalid
-      if (!std::isfinite ((*input_)[*it].x) ||
-          !std::isfinite ((*input_)[*it].y) ||
-          !std::isfinite ((*input_)[*it].z))
+      if (!isXYZFinite ((*input_)[index]))
         continue;
 
-    int ijk0 = static_cast<int> (std::floor ((*input_)[*it].x * inverse_resolution_) - static_cast<float> (min_b[0]));
-    int ijk1 = static_cast<int> (std::floor ((*input_)[*it].y * inverse_resolution_) - static_cast<float> (min_b[1]));
+    int ijk0 = static_cast<int> (std::floor ((*input_)[index].x * inverse_resolution_) - static_cast<float> (min_b[0]));
+    int ijk1 = static_cast<int> (std::floor ((*input_)[index].y * inverse_resolution_) - static_cast<float> (min_b[1]));
 
     // Compute the grid cell index
     int idx = ijk0 * divb_mul[0] + ijk1 * divb_mul[1];
-    index_vector.emplace_back(static_cast<unsigned int> (idx), *it);
+    index_vector.emplace_back(static_cast<unsigned int> (idx), index);
```

Comment:
I think modification of `ijk{0,1}`, `idx` and `point_index_idx` is also needed, else you're causing an overflow in the first ones and then other variables just have garbage in them (for index_t == int64_t)

Reply:
Maybe we can leave that for the future? Like I wrote above, this PR is only one step, to do the most obvious changes. I think a detailed analysis of types is needed in grid_minimum.hpp (and other files), since there is a lot of casting between `int`, `unsigned int`, and `float`.